golang/cosmos/app/app.go

```diff
@@ -358,7 +358,12 @@ func NewAgoricApp(
 	)
 
 	// This function is tricky to get right, so we build it ourselves.
-	callToController := func(ctx sdk.Context, str string) (string, error) {
+	callToController := func(ctx sdk.Context, str, simReturn string) (string, error) {
+		if vm.IsSimulation(ctx) {
+			// Just return the simReturn, since the message is being simulated.
+			return simReturn, nil
```

Comment:
It sure looks like `simReturn` is always the empty string. If so, you can avoid the extra arg to `callToController` and avoid a lot of changes.